utils/s2n_array.c

```diff
@@ -156,6 +176,7 @@ S2N_RESULT s2n_array_remove(struct s2n_array *array, uint32_t idx)
                    0,
                    array->element_size);
 
+    RESULT_GUARD(s2n_array_validate(array));
```

Comment:
Not sure if this validation function will show up in a profile but generally any validation we do after a mutation uses `RESULT_POSTCONDITION` instead. This will only validate in debug builds. We use this all over the place in blob and stuffer since the validation function was showing up in profiling.

Not saying we need to switch here; it's just something to keep in mind.

Reply:
Fixed! e3f0784aa9929e9d78af209d900946f12b9a5d86

Reply:
Fixed again: 48816166f1eb5e5d10cbadca7ea9329fd0881cf8

Reply:
I would have said swapping all the existing validates would have been out of scope, so even nicer cleanup!